core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java

```diff
@@ -387,6 +376,84 @@ public void onViewUpdated(final ViewMetadata current, final ViewMetadata previou
         // NOOP
     }
 
+    /**
+     * Called when the session is up and ready. Will invoke the listeners' onSessionReady methods.
+     *
+     * @param session The session
+     */
+    @Override
+    public void onSessionReady(final Session session)
+    {
+        SchemaChangeListener.super.onSessionReady(session);
+        // prepare the NodeStateListener
+        NodeStateListener.super.onSessionReady(session);
+    }
+
+    /**
+     * Callback for when a node is added to the cluster.
+     *
+     * @param node
+     */
+    @Override
+    public void onAdd(final Node node)
+    {
+        setupConfiguration();
+    }
+
+    /**
+     * Callback for when a node is considered to be up.
```

Comment:
I guess it is "changes state to up"? It's not called while a node is already up, right?

Reply:
Yes, that's right. It is called when an actual connection is established with the node. That is, the driver then considers the node to be up.

Reply:
Changed to "... switches state ..."